README.md

```diff
@@ -116,6 +116,33 @@ fun setup() {
 }
 ```
 
+JaCoCo Compatibility
+--------------------
+
+The way both JaCoCo and Paparazzi instrument may cause conflict if JDK 17+ is used. Mitigate this issue by adding the following into your JaCoCo configuration [#955](https://github.com/cashapp/paparazzi/issues/955):
+
+build.gradle
+```groovy
+jacoco.excludes = [
```

Comment:
I think Gradle Groovy uses a weird reflection API. If a function is not found in a class, it automatically finds it in some extension. I tried the similar but it is definitely not possible in Kotlin. You need to find the "extension" and configure explicitly.

Reply:
Yeah, `GroovyObject` and friends, which results in an elaborate way to resolve properties:
https://docs.gradle.org/current/dsl/org.gradle.api.Project.html#org.gradle.api.Project:property(java.lang.String)

---

I had a deeper look because I was curious:
```gradle
// build.gradle
plugins {
	id("jacoco")
}
println(jacoco) // extension 'jacoco'
println(jacoco.class) // class org.gradle.testing.jacoco.plugins.JacocoPluginExtension_Decorated
jacoco.excludes = [] // fails with:
// > Could not set unknown property 'excludes' for extension 'jacoco' of type org.gradle.testing.jacoco.plugins.JacocoPluginExtension.
```
Same output in `.gradle.kts` except the `.excludes` is a compile error, because that property only exists on `JacocoTaskExtension`.

To access the right "jacoco" we still need to qualify with a task:
```suggestion
tasks.test.jacoco.excludes = [ ...
```
or
```suggestion
tasks.withType(Test).configureEach {
    jacoco.excludes = [ ...
```

---

Btw, the same is possible for Project-level extension in Kotlin thanks to generated accessors. The difference is that `tasks.test` in Groovy will give us a materialized `Task`, while in Kotlin it's defined as `TaskProvider<Test>` and uses `named()`. Sadly this is a task-level extension so we need to use the extended kts syntax to find the extension as you did.

Reply:
Ok. This makes a lot of sense. Thanks for the enlightening comment.

I was confused by 2 extensions. Jacoco has 2 different plugin and task extensions with the same name 😅 I copied the groovy code piece from the jacoco repo linked above.

I guess your first suggestion is the best because it is simpler. Every project has a different style of configuring these so anyone reading this needs to adjust slightly.